src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml

```diff
@@ -1603,7 +1606,21 @@ module Circuit = struct
   let () =
     circuit##.runAndCheck :=
       Js.wrap_callback (fun (f : unit -> 'a) ->
-          Impl.run_and_check (fun () -> f) |> Or_error.ok_exn ) ;
+          (* TODO: this is using `run_and_check` wrong. Instead, it should use `run_circuit` and wrap on the JS side to read return values *)
+          let result =
+            try Impl.run_and_check (fun () -> f) with exn -> raise_exn exn
```

Comment:
Don't you want `run_and_check_exn` here too?

Reply:
yea you're right!